### PR TITLE
Use transactions to create pacemaker resources

### PR DESCRIPTION
**Pull request by vuntz**
This depends on https://github.com/crowbar/crowbar-ha/pull/57

### CODE REVIEW

**Comment by jsuchome** [comment]
+1

File: chef/cookbooks/ceph/recipes/radosgw_ha.rb [hunk 1]
```diff
@@ -31,18 +31,30 @@
...
 
+service_name = "ceph-radosgw"
 pacemaker_primitive service_name do
   agent node[:ceph][:ha][:radosgw][:agent]
   op node[:ceph][:ha][:radosgw][:op]
-  action :create
+  action :update
   only_if { CrowbarPacemakerHelper.is_cluster_founder?(node) }
 end
+transaction_objects << "pacemaker_primitive[#{service_name}]"
```
Comment (rsalevsky) [inline chef/cookbooks/ceph/recipes/radosgw_ha.rb:43]: `transaction_objects.push("pacemaker_primitive[#{service_name}]")`

File: chef/cookbooks/ceph/recipes/radosgw_ha.rb [hunk 2]
```diff
@@ -31,18 +31,30 @@
...
-pacemaker_clone "cl-#{service_name}" do
+clone_name = "cl-#{service_name}"
+pacemaker_clone clone_name do
   rsc service_name
-  action [:create, :start]
+  meta ({ "clone-max" => CrowbarPacemakerHelper.num_corosync_nodes(node) })
+  action :update
+  only_if { CrowbarPacemakerHelper.is_cluster_founder?(node) }
+end
+transaction_objects << "pacemaker_clone[#{clone_name}]"
```
Comment (rsalevsky) [inline chef/cookbooks/ceph/recipes/radosgw_ha.rb:52]: `transaction_objects.push("pacemaker_clone[#{clone_name}]")`

File: chef/cookbooks/ceph/recipes/radosgw_ha.rb (hunk 1 above)
Comment (mjura) [inline chef/cookbooks/ceph/recipes/radosgw_ha.rb:43]: Why don't want to use `<<` operator? Codeclimate and hound are OK with this code.

File: chef/cookbooks/ceph/recipes/radosgw_ha.rb (hunk 2 above)
Comment (mjura) [inline chef/cookbooks/ceph/recipes/radosgw_ha.rb:52]: Why don't want to use `<<` operator? Codeclimate and hound are OK with this code.

File: chef/cookbooks/ceph/recipes/radosgw_ha.rb [hunk 3]
```diff
@@ -31,18 +31,30 @@
...
   only_if { CrowbarPacemakerHelper.is_cluster_founder?(node) }
 end
+transaction_objects << "pacemaker_primitive[#{service_name}]"
 
-pacemaker_clone "cl-#{service_name}" do
+clone_name = "cl-#{service_name}"
+pacemaker_clone clone_name do
   rsc service_name
-  action [:create, :start]
+  meta ({ "clone-max" => CrowbarPacemakerHelper.num_corosync_nodes(node) })
```
Comment (mjura) [inline chef/cookbooks/ceph/recipes/radosgw_ha.rb:48]: `meta ({ "clone-max" => CrowbarPacemakerHelper.num_corosync_nodes(node) })` codelimate is also unhappy with this

Comment (vuntz) [inline chef/cookbooks/ceph/recipes/radosgw_ha.rb:48]: I think it's a bug in rubocop. At least, I can't find a way to make that work (removing `()` doesn't work, removing `{}` doesn't work, etc.)

File: chef/cookbooks/ceph/recipes/radosgw_ha.rb (hunk 2 above)
Comment (tboerger) [inline chef/cookbooks/ceph/recipes/radosgw_ha.rb:52]: @mjura because we are using push on most places. This can also getting better arranged.

**Comment by aplanas** [comment]
gating fail: one machine never goes to ready state, so fails too early to test it

File: chef/cookbooks/ceph/recipes/radosgw_ha.rb (hunk 2 above)
Comment (mjura) [inline chef/cookbooks/ceph/recipes/radosgw_ha.rb:52]: Thanks for explanation, I'm fine

**Comment by vuntz** [comment]
Any more reviews on this?

**Comment by aspiers** [comment]
+1

**Pull request merged**